### PR TITLE
Create a dedicated upgrade command

### PR DESCRIPTION
**Pull request by stnguyen90**
## What does this PR do?

Before, we used the same command for both installation and upgrades. This lead to problems because developers would try to upgrade in the wrong folder and end up creating a new installation.

This new upgrade command validates the existence of an existing installation before proceeding with the upgrade to ensure no new installation is created when upgrading.

## Test Plan

The following screenshot shows:

1. Running the upgrade command w/o an installation results in an error
2. After adding an installation (appwrite folder w/ docker-compose.yml file), the install command runs as usual
   * The error at the end occurs because I'm running in an appwrite instance where the docker socket is unavailable. When someone tries to upgrade, they will have the docker socket available and this error won't show.

Screen Shot 2023-07-20 at 7 11 29 PM

Latest error output:

image

## Related PRs and Issues

- https://github.com/appwrite/appwrite/issues/2616

## Checklist

- [x] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [x] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

**Review by eldadfux — COMMENTED** [review]
@stnguyen90 this is not exactly how we use class based tasks, we've created a new abstraction layer for them here: https://github.com/appwrite/cloud/tree/feat-billing/src/Appwrite/Cloud/Platform/Tasks we might be able to reuse this logic as we need to rewrite them anyway.

Please check with @lohanidamodar if we can adapt this approach here.

File: src/Appwrite/Task/Setup.php [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    }
+
+    public function upgrade()
+    {
+        // Check for previous installation
+        $data = @file_get_contents(SETUP::INSTALLATION_PATH . '/docker-compose.yml');
+        if (empty($data)) {
+            Console::error('Appwrite installation not found.');
+            Console::log('The command was not run in the parent folder of your appwrite installation.');
+            Console::log('Please navigate to the parent of the Appwrite installation and try again.');
```
Comment (gewenyu99) [inline src/Appwrite/Task/Setup.php:262]:
```suggestion
            Console::log('Please navigate to the parent directory of the Appwrite installation and try again.');
```

**Review by gewenyu99 — CHANGES_REQUESTED** [review]
LGTM, but I want to add one word to feel useful.

(Jokes aside I think we should be verbose here.

**Comment by stnguyen90** [comment]
Need to restructure the task.

**Pull request closed without merging**